Feature/mcw 34 discounts

Yellowatch opened the pull request:
Connected coupons and promotions with django. Admin can choose which promotion to display, if it needs user to be logged in, and if user must have not made any purchases yet.
![image](https://github.com/G2-Magnolia-Cakes-Website/Magnolia-Cakes-Website/assets/63491152/32608a7c-1099-4817-a91a-a22dccdbd48c)
![image](https://github.com/G2-Magnolia-Cakes-Website/Magnolia-Cakes-Website/assets/63491152/eb3d0231-3206-4023-9923-16fb561d82bc)

Yellowatch committed:
testing stripe

Yellowatch committed:
Merge branch 'develop' into feature/MCW-34-Discounts

Yellowatch committed:
Building the coupon and promotion coupons

Yellowatch committed:
fix views

Yellowatch committed:
Fixed the models for coupons and promotions

Yellowatch committed:
Completed Displayed Promotion

Yellowatch committed:
Get basic frontend popup working

Yellowatch committed:
Frontend Promotion popup

Yellowatch committed:
Option to only show promotion to logged in users

Yellowatch committed:
Fix login popup not showing issue

Yellowatch committed:
Merge branch 'develop' into feature/MCW-34-Discounts

Yellowatch committed:
make migrations

Yellowatch committed:
Add feature only first order promotion

quangminhduong reviewed (approved):
Great work, can you add this line onto the create_checkout_session method and see if work too? I tried and it actually apply the promotion to the payment

![image](https://github.com/G2-Magnolia-Cakes-Website/Magnolia-Cakes-Website/assets/49256247/e9a42be0-340f-4a7e-a274-409b0d6691dd)

Yellowatch committed:
Added time for promotion, and fixed backend

Yellowatch committed:
Merge branch 'develop' into feature/MCW-34-Discounts

Yellowatch committed:
Fixed migration issues

Yellowatch commented:
New changes:

- Fixed purchasing bugs
- Fixed styling
- added "added to cart" success message to both workshop and online store

![image](https://github.com/G2-Magnolia-Cakes-Website/Magnolia-Cakes-Website/assets/63491152/9eaa8fcf-c8f1-4488-baaa-a40a1ef1616b)

![image](https://github.com/G2-Magnolia-Cakes-Website/Magnolia-Cakes-Website/assets/63491152/bf501dca-4491-4f45-af1a-71119edd229a)

![image](https://github.com/G2-Magnolia-Cakes-Website/Magnolia-Cakes-Website/assets/63491152/85dbf8fb-fc82-44d6-add8-b4ef0c2b7806)

Yellowatch committed:
Merge branch 'develop' into feature/MCW-34-Discounts

Yellowatch committed:
makemigrations

Yellowatch committed:
Fixed code merge fail

Yellowatch committed:
Fixed workshop style

Yellowatch committed:
Add not logged in page to workshop and onlinestore
